dashboard/src/app/workspaces/workspace-details/workspace-details.controller.js

```diff
@@ -96,36 +96,100 @@ export class WorkspaceDetailsCtrl {
     }
     this.workspaceId = this.workspaceDetails.id;
     this.newName = angular.copy(this.workspaceDetails.config.name);
+    this.newRam = this.getRam();
   }
 
-  //Rename the workspace.
-  renameWorkspace() {
-    this.isLoading = true;
-
-    let promise = this.cheWorkspace.fetchWorkspaceDetails(this.workspaceId);
+  /**
+   * Returns amount of RAM for dev machine in default environment.
+   * @returns {*}
+   */
+  getRam() {
+    // get default environment
+    let defaultEnv = this.lodash.find(this.workspaceDetails.config.environments, (env) => {
+      return env.name === this.workspaceDetails.config.defaultEnv;
+    });
 
-    promise.then(() => {
-      this.doRenameWorkspace();
-    }, () => {
-      this.doRenameWorkspace();
+    // get dev machine config
+    let devMachineConfig = this.lodash.find(defaultEnv.machineConfigs, (machine) => {
+      return machine.dev === true;
     });
+
+    return angular.copy(devMachineConfig.limits.ram);
+  }
+
+  /**
+   * Returns true if name of workspace is changed.
+   * @returns {boolean}
+   */
+  isNameChanged() {
+    if (this.workspaceDetails) {
+      return this.workspaceDetails.config.name !== this.newName;
+    }
+    return false;
   }
 
-  doRenameWorkspace() {
+  /**
+   * Returns true if amount of RAM of workspace is changed
+   * @returns {boolean}
+   */
+  isRamChanged() {
+    if (this.workspaceDetails) {
+      return this.getRam() !== this.newRam;
+    }
+    return false;
+  }
+
+  /**
+   * Calls method to update workspace info after timeout.
+   * @param isFormValid {Boolean} true if form is valid
+   */
+  updateWorkspace(isFormValid) {
+    this.$timeout.cancel(this.timeoutPromise);
+
+    if (isFormValid === false || !(this.isNameChanged() || this.isRamChanged())) {
+      return;
+    }
+
+    this.timeoutPromise = this.$timeout(() => {
+      this.isLoading = true;
+      this.cheWorkspace.fetchWorkspaceDetails(this.workspaceId).then(() => {
+        this.doUpdateWorkspace();
+      }, () => {
+        this.doUpdateWorkspace();
+      });
+    }, 2000);
+  }
+
+  /**
+   * Updates workspace info.
+   */
+  doUpdateWorkspace() {
     this.workspaceDetails = this.cheWorkspace.getWorkspacesById().get(this.workspaceId);
     let workspaceNewDetails = angular.copy(this.workspaceDetails);
+
     workspaceNewDetails.config.name = this.newName;
+
+    workspaceNewDetails.config.environments.forEach((env) => {
```

Comment:
lodash can be used here same as https://github.com/eclipse/che/pull/2061/files#diff-62c417fbca8214333da4620deda20be7R108